certGenerator.go

```diff
@@ -0,0 +1,46 @@
+// based on https://stackoverflow.com/q/26441547/1426535
+
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"io/ioutil"
+	"math/big"
+	"time"
+)
+
+func main() {
+
+	var filename = "test"
+
+	template := x509.Certificate{
+		SerialNumber:   big.NewInt(time.Now().Unix()),
+		Subject:        pkix.Name{CommonName: "example"},
+		NotBefore:      time.Date(2021, time.January, 31, 0, 0, 0, 0, time.UTC),
+		NotAfter:       time.Date(2021, time.January, 31, 0, 0, 0, 0, time.UTC).AddDate(1, 0, 0),
+		KeyUsage:       x509.KeyUsageKeyEncipherment,
+		ExtKeyUsage:    []x509.ExtKeyUsage{x509.ExtKeyUsageEmailProtection},
+		EmailAddresses: []string{"test@example.com"},
+	}
+
+	privatekey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		panic(err)
+	}
+
+	crt, err := x509.CreateCertificate(rand.Reader, &template, &template, &privatekey.PublicKey, privatekey)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
I have to say, I'm a little bit torn about whether this is really a useful piece of ZLint documentation.

As one example, in my experience writing other small programs to generate ZLint test certificates (e.g.[sctTestCerts.go](https://gist.github.com/cpu/6d26b2718f29e184ff88a90f02d7cbcb/raw/2a5f8a0c673f197de83d176667c76f5f8e498147/sctTestCerts.go)) I've needed to do more contortions to make sure the test certs will be linted. Anything following this example closely will never pass the (very commonly used) [`util.IsSubscriberCert`](https://github.com/zmap/zlint/blob/848c50b07013a773f2d98620400a6be900c88d2c/v3/util/ca.go#L41-L45) check because the example is a self-signed certificate. You could fix the example to also generate an issuer certificate but then it will be even more code in-tree.